src/main/java/ru/practicum/shareit/item/dto/ItemMapper.java

```diff
@@ -1,13 +1,33 @@
 package ru.practicum.shareit.item.dto;
 
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.springframework.beans.factory.annotation.Autowired;
+import ru.practicum.shareit.booking.service.BookingService;
 import ru.practicum.shareit.item.model.Item;
+import ru.practicum.shareit.item.storage.CommentRepository;
 
-@Mapper
-public interface ItemMapper {
-    ItemDto itemToDto(Item item);
+@Mapper(componentModel = "spring")
```

Comment:
Раньше не сталкивался с этим. Оно действительно удобно? Не тормозит? 
Тут, похоже, реализация через отражения сделана...

Reply:
Проблем со временем работой маппера mapstruct при изменении с интерфейса на абстрактный класс не заметила.
Я нашла этот механизм внедрения и использования методов бинов  на сайте гида по Spring  (https://www.baeldung.com/mapstruct)

5.2. Inject Spring Components into the Mapper
Sometimes, we'll need to utilize other Spring components inside our mapping logic. In this case, we have to use an abstract class instead of an interface:

@Mapper(componentModel = "spring")
public abstract class SimpleDestinationMapperUsingInjectedService

Then, we can easily inject the desired component using a well-known @Autowired annotation and use it in our code:

@Mapper(componentModel = "spring")
public abstract class SimpleDestinationMapperUsingInjectedService {

    @Autowired
    protected SimpleService simpleService;

    @Mapping(target = "name", expression = "java(simpleService.enrichName(source.getName()))")
    public abstract SimpleDestination sourceToDestination(SimpleSource source);
}
Copy
We must remember not to make the injected bean private! This is because MapStruct has to access the object in the generated implementation class.

Reply:
Но про отражение сейчас не скажу. Начала читать, для меня это пока темный лес :((